zipkin-storage/elasticsearch/src/main/java/zipkin/storage/elasticsearch/InternalElasticsearchClient.java

```diff
@@ -85,32 +117,18 @@
    * NB: This call is _lenient_ in its index selection, and only will match open indexes, cf.
    * {@link IndicesOptions#lenientExpandOpen()} }.
    */
-  ListenableFuture<Collection<DependencyLink>> findDependencies(String[] indices);
-
-  /**
-   * Indexes the given dependency links, flushing unconditionally.
-   */
-  void indexDependencies(String index, List<IndexableLink> links);
-
-  final class IndexableLink {
-    final String id;
-    final Map<String, ?> data;
-
-    public IndexableLink(String id, Map<String, ?> data) {
-      this.id = id;
-      this.data = data;
-    }
-  }
+  // TODO(adrian): revisit once the other implementation is in: why we are using arrays vs lists?
+  protected abstract ListenableFuture<Collection<DependencyLink>> findDependencies(String[] indices);
```

Comment:
The java elasticsearch client API uses varargs for its index lists; this allows a direct pass-through on that stuff and a zero-copy single-allocation wrapper when we need a list (i.e. `Arrays#asList`).

Alternatively, we can take a list here, but then we'd have to do the `indices.toArray(new String[indices.size()])` dance before handing them off to the elasticsearch client.